src/device-api/v1.ts

```diff
@@ -218,25 +211,10 @@ router.patch('/v1/device/host-config', async (req, res) => {
 	}
 
 	try {
-		// If hostname is an empty string, return first 7 digits of device uuid
-		if (req.body.network?.hostname === '') {
-			const uuid = await config.get('uuid');
-			req.body.network.hostname = uuid?.slice(0, 7);
-		}
-		const lockOverride = await config.get('lockOverride');
-		await hostConfig.patch(
-			req.body,
-			checkTruthy(req.body.force) || lockOverride,
-		);
-		res.status(200).send('OK');
-	} catch (err: any) {
-		// TODO: We should be able to throw err if it's UpdatesLockedError
-		// and the error middleware will handle it, but this doesn't work in
-		// the test environment. Fix this when fixing API tests.
-		if (err instanceof UpdatesLockedError) {
-			return res.status(423).send(err?.message ?? err);
-		}
-		res.status(503).send(err?.message ?? err ?? 'Unknown error');
```

Comment:
The error handling middleware handles these errors now, so the 503 message returned in this case will be a JSON message `{ status: 'failed', message: err?.message ?? err ?? 'Unknown error' }` instead of what's sent here.

Reply:
I think that's a good improvement

Reply:
Actually, on second though, maybe this would be considered a breaking change, since someone may have a device that is depending on the response format. The break depends on the user updating the supervisor, but still is probably not great if they update their supervisor on an old device and suddenly their app is malfunctioning.

Maybe let's open a discussion Zulip about what would be the considered breaking changes on an API like the supervisor's. What sort of changes depend on an API version change and what depend on a Supervisor version change

My guesses

- I think probably a change in an error response is probably OK (e.g. 400 ->  404)
- Maybe a change in an error message is fine (e.g. 'Unknown error' -> 'An exception occurred when trying to ...')
- A format change in error response is probably not OK, but maybe we can get away with a Supervisor version bump
- A non additive change in a 2xx response is a breaking change and should not be done outside of an API version bump

Reply:
We discussed this in a Supervisor checkin and agreed that all interfaces should stay the same, with the exception of string contents. So, error codes remain as-is even if incorrect, JSON responses retain their keys (although new keys can be added), although messages, whether as text responses or JSON values, can change. I'll revert this one case to the interface it was before to keep it consistent.